integration_tests/src/fixture_generator/reader/yaml.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under both the MIT license found in the
+// LICENSE-MIT file in the root directory of this source tree and the Apache
+// License, Version 2.0 found in the LICENSE-APACHE file in the root directory
+// of this source tree.
+
+//! This module contains an implementor of the Reader trait for the YAML format.
+
+use std::fs::File;
+use std::io::{BufRead, BufReader, Lines, Seek, SeekFrom};
+use std::iter::Peekable;
+
+use serde::de::DeserializeOwned;
+
+use crate::fixture_generator::generator::{Delta, Metadata, State};
+use crate::fixture_generator::reader::Reader;
+
+const YAML_SEPARATOR: &str = "---";
+
+/// YAML format file reader.
+pub struct YamlFileReader {
+    file: File,
+    index: u32,
+    buffer: Peekable<Lines<BufReader<File>>>,
+}
+
+impl YamlFileReader {
+    pub fn new(file: File) -> Self {
+        let index = 0;
+        let buffer = Self::buffer(&file);
+        Self {
+            file,
+            index,
+            buffer,
+        }
+    }
+
+    // Instantiates a new buffer for a given file.
+    fn buffer(file: &File) -> Peekable<Lines<BufReader<File>>> {
+        let mut file_ref_copy = file.try_clone().unwrap();
+        file_ref_copy.seek(SeekFrom::Start(0)).unwrap();
+
+        BufReader::new(file_ref_copy).lines().peekable()
+    }
+
+    // Returns the next YAML "doc" in the file, looping back to the start of the
+    // file if EOF is encountered.
+    fn next_doc(&mut self) -> String {
+        // find start of doc
+        loop {
+            match self.buffer.peek() {
```

Comment:
Ah, we're moving forward line by line thanks to the `.lines()` call here:

https://github.com/novifinancial/akd/blob/573f3cc5589b246f9ce31e47482da23ff8a5aefe/integration_tests/src/fixture_generator/reader/yaml.rs#L44

Reply:
so then the entire line (incl whitespace) needs to be "---" right? Maybe we should trim this line to remove whitespace?

Since they're human readable, they're also human editable lol (i.e. human breakable)

Reply:
That's a good call! I've added a trim function but only for the end of the string, whitespace before `---` is considered invalid YAML.

Reply:
That's a good call! I've added a trim function but only for the end of the string, whitespace before `---` is considered invalid YAML.